### PR TITLE
Add option to specify a github-token to avoid rate-limit

### PR DESCRIPTION
**Pull request by maginseb**
Hi, 
I hope it is okay for me to just open a pull request with a proposal. I'm running this action on a GitHub Enterprise machine. The problem that I'm facing is that ~50% of the times the request to check for the latest version of Piper gets rejected by GitHub due to rate-limiting. I found that if I authenticate to github.com I could work around this rate-limit and not run into the issue.

### CODE REVIEW

**Comment by maginseb** [comment]
> [CLA assistant check](https://cla-assistant.io/SAP/project-piper-action?pullRequest=177) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/SAP/project-piper-action?pullRequest=177) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/SAP/project-piper-action?pullRequest=177) it.

It seems that I can't sign the CLA unfortunately. The bot sends me to an empty page.

**Comment by CCFenner** [comment]
Thanks for the PR, unfortunately this action is deprecated and we're working on the `main` branch. Can you switch to target that branch?

File: README.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 [![REUSE status](https://api.reuse.software/badge/github.com/SAP/project-piper-action)](https://api.reuse.software/info/github.com/SAP/project-piper-action)
 
+:construction: The master branch is soon to be deprecated. The main branch will become the default branch.
+
```
Comment (CCFenner) [inline README.md:6]:
```suggestion
```

**Comment by CCFenner** [comment]
@maginseb can you resolve the conflict?

**Comment by maginseb** [comment]
@CCFenner My mistake. I found that what I was trying to implement already exists in the `main` branch. Sorry for my mistake, I didn't realise that the master branch was heavily outdated.

For everyone who sees this and is currently facing the issue of 403s on GitHub Enterprise:

Be sure to hand over an access token to github.com using the input `github-token`. This will prevent you from running into the rate-limit.

**Pull request closed without merging**